### PR TITLE
21 merge lists

### PR DESCRIPTION
**Pull request by oleggromov**
This is a solution to [21 Merge Two Sorted Lists](https://leetcode.com/problems/merge-two-sorted-lists)

## Description
The first thought I had and actually the one I've implemented is trivial: 
- create a new list 
- advance through the two given lists with a while loop and 2 pointers
- apply the least element as the next

I had to cover an edge-case when one list is over, meaning that we need to link the rest of the other to the end of our current.

### Attempts to make it more efficient

So initially it was a single while loop with more complicated conditionals so that they would work for both normal and edge cases.

I later realized I don't have to loop over the rest of the remaining linked list to apply elements one-by-one and instead just link the rest to the last element directly.

Somehow, this direct solution made it _less performant_, which comes at a surprise to me... See the submissions above the first one (11 hours ago). 

Screenshot 2019-09-05 at 22 48 13

Finally, I realized that instead of having `if-else if` I can go over 2 elements from both lists at a single iteration, which got me back to 60ms execution time.

### What could be done better
- In terms of time complexity, I honestly don't know. Seems that you have to iterate over the two lists...
- In terms of space, I think I may not need to create new elements but instead cross-link existing ones. 

IMG_8408

WDYT, Viktor? Please don't give me exact solutions but instead a few hints on what I could aim for in terms of results and which directions to take. Thank you 🤘

### CODE REVIEW

**Comment by oleggromov** [comment]
Looking at the code, I realize that I already effectively cross-link the existing elements.
Any thoughts, Viktor?

File: 21-merge-lists.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  let current = fakeHead
+
+  while (l1 && l2) {
+    if (l1.val <= l2.val) {
+      current.next = l1
+      current = current.next
+      l1 = l1.next
+    }
+
+    if (l1 && l2.val <= l1.val) {
```
Comment (vitkarpov) [inline 21-merge-lists.js]: I'd use `else` here, so there's no need to check `l1` once again.

**Comment by vitkarpov** [comment]
> Somehow, this direct solution made it less performant

I wouldn't trust LeetCode here much, usually, it gives significant inaccuracy.

**Comment by vitkarpov** [comment]
For instance, I just submitted two identical solutions in a row:

Screen Shot 2019-09-06 at 8 12 17 AM

`¯\_(ツ)_/¯`

File: 21-merge-lists.js (hunk 1 above)
Comment (oleggromov) [reply 21-merge-lists.js]: Yes, you're right. My thinking was "we will use less iterations" but it's exactly the same to make an iteration over again or get into another conditional. Will do.

**Comment by oleggromov** [comment]
Thanks, @vitkarpov! I'll merge now.

**Pull request merged**